hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java

```diff
@@ -3577,18 +3578,24 @@ public List<OzoneAcl> getAcl(OzoneObj obj) throws IOException {
    * @return If checkpoint is installed successfully, return the
    *         corresponding termIndex. Otherwise, return null.
    */
-  public TermIndex installSnapshotFromLeader(String leaderId) {
+  public synchronized TermIndex installSnapshotFromLeader(String leaderId) {
     if (omRatisSnapshotProvider == null) {
       LOG.error("OM Snapshot Provider is not configured as there are no peer " +
           "nodes.");
       return null;
     }
 
+    java.util.Optional<SstFilteringService> sstFilteringService =
+        java.util.Optional.ofNullable(
+            keyManager.getSnapshotSstFilteringService());
     DBCheckpoint omDBCheckpoint;
+    sstFilteringService.ifPresent(BackgroundService::shutdown);
```

Comment:
I don't think it is a good idea to shutdown and start the service like this. I think we should use lock as it is done in other places like following:
* https://github.com/apache/ozone/blob/0862a76ef178f427a272c335374760c87c885119/hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/RocksDBCheckpointDiffer.java#L1167
* https://github.com/apache/ozone/blob/1ae0401b1ac662c021b37b68ef46f4b9a0909890/hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/DBCheckpointServlet.java#L174

@GeorgeJahad can add or correct me if we can't use that.

I see we have a lock: https://github.com/apache/ozone/blob/e950ecee512fd87a9c20c233ba6ed224acd2c6e4/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/SstFilteringService.java#L164

We may need to place this lock before starting the SST file cleanup.

Reply:
We are shutting down the service anyway before installation. Locking doesn't make sense.

Reply:
yes, I believe the sst filtering service is already being stopped here: https://github.com/apache/ozone/blob/ffd84e747dc43d08a23269a8f411d52fef41c432/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java#L3640  Why do we need to do it again?

Reply:
Had an offline discussion with Swami and Prashant. Race condition Swami is trying to solve here is when follower requests leader for tarball (incremental diff) with excluded SST files list based on current state of the follower, leader will only send the diff SST files and new manifest file. In meanwhile SST Filtering service might delete irrelevant SST files. Follower loads DB based on new manifest from leader and  when it does that some files might not existed on follower because SST Filtering service has removed them in between fetching and loading. This leaves DB (manifest and data) into inconsistent state. Because manifest contains SST files which actual doesn't exist on follower.

I hope this help @GeorgeJahad.

@swamirishi added some details here: https://github.com/apache/ozone/pull/5035/#issuecomment-1629504655